src/cache.jl

```diff
@@ -1,40 +1,3 @@
-
-function Base.getproperty(cache::SciMLBase.AbstractOptimizationCache, x::Symbol)
-    if x in (:u0, :p)
-        return getfield(cache.reinit_cache, x)
-    end
-    return getfield(cache, x)
-end
-
-SciMLBase.has_reinit(cache::SciMLBase.AbstractOptimizationCache) = true
-function SciMLBase.reinit!(cache::SciMLBase.AbstractOptimizationCache; p = missing,
```

Comment:
Why did you delete this?

Reply:
This is type-piracy. They should be defined in SciMLBase, and have been moved there in https://github.com/SciML/SciMLBase.jl/pull/603

Reply:
If there isn't a dispatch defined there this isn't type piracy (?) but that's fine, it does make sense to move it. I'll just setup OptimizationBase and move all these things there, that seems necessary with the number of structs being used here now and the methods associated with them.

Reply:
> If there isn't a dispatch defined there this isn't type piracy (?)

It is type piracy. It's not whether dispatches exist, usually they don't, it's what owns the function and types. In this case, the abstract type and the function are both defined in SciMLBase, so indeed it is type piracy. 

But yes, a base level for this kind of stuff is kind of required. Optimization.jl has a very mixed identity as a package right now, supporting both AD and being a solver front end.